debugging/src/buggy.cpp

```diff
@@ -33,5 +33,6 @@ void printStudent(Student* myStudent){
 int main(){
   Student* myStudent = getStudent();
   printStudent(myStudent);
+  delete myStudent;
```

Comment:
you could have used a smart pointer here.

Reply:
Is there a way to do this without overwriting all standard pointer usage in the code? Initialising as a smart pointer type breaks `printStudent` as it expects a raw pointer as input.

Reply:
This may help: http://en.cppreference.com/w/cpp/memory/unique_ptr/get